src/simulator/world.py

```diff
@@ -170,7 +172,18 @@ def visualize(self) -> dict:
         """
         objects_info = dict()
         for oid in self.objects:
-            if self.objects[oid].shape is not None:
                 # TODO: to be replaced with proper handling of compound objects (if cond)
-                objects_info[oid] = self.objects[oid].visualize()
+            objects_info[oid] = self.objects[oid].visualize()
```

Comment:
If this line is not part of the above `if`, then what is that `if` actually doing now?